src/node-api/v8_api.cpp

```diff
@@ -942,3 +945,267 @@ napi_create_external_buffer(napi_env env,
   *result = v8impl::JsValueFromV8LocalValue(buffer);
   return GET_RETURN_STATUS(env);
 }
+
+// Helper functions for instrumentation APIs
+namespace {
+
+facebook::jsi::Instrumentation* GetInstrumentation(napi_env env) {
+  auto v8impl_env = static_cast<v8impl::NodeApiEnv*>(env);
+  if (!v8impl_env) {
+    return nullptr;
+  }
+  
+  v8::Isolate* isolate = v8impl_env->isolate;
+  if (!isolate) {
+    return nullptr;
+  }
+  
+  // Create a new V8Instrumentation instance just for handling this call
+  // This gives us access to the V8 instrumentation capabilities without 
+  // needing to access the private runtime member
+  return new v8runtime::V8Instrumentation(isolate);
+}
+
+napi_status CopyStringToOutput(napi_env env, const std::string& str, char** result, size_t* result_length) {
```

Comment:
While this pattern works, it causes unnecessary string allocations in almost all scenarios.
It would be better to use API that receives a callback instead.
E.g. see the new JSI API to retrieve string data: https://github.com/facebook/hermes/blob/11c154bc365447648fa56fdd67a1c142034b85a6/API/jsi/jsi/jsi.h#L424

Reply:
Fixed